src/training/train.py

```diff
@@ -37,6 +37,26 @@ def update(self, val, n=1):
         self.avg = self.sum / self.count
 
 
+class Batch(dict):
+    """Batch wrapper for a dictionary of tensors."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(self, *args, **kwargs)
+
+    def size(self):
+        if self.keys():
+            k = list(self.keys())[0]
+            return len(self[k])
+        return 0
+
+    def to(self, *, device, non_blocking=False, dtypes={}):
+        for k, v in self.items():
+            if isinstance(v, torch.Tensor):
+                dtype = dtypes.get(k, v.dtype)
+                self[k] = v.to(device=device, non_blocking=non_blocking, dtype=dtype)
```

Comment:
You can probably recursively call it on dicts as well. I am not sure what level of compliance you would want to go towards.